ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/anomalylocalization/AnomalyLocalizerImpl.java

```diff
@@ -150,6 +153,20 @@ private void onOverallAggregatesResponse(MultiSearchResponse response, AnomalyLo
             listener.onFailure(new IndexNotFoundException("Failed to find index: " + input.getIndexName()));
         }
     }
+    
+    protected boolean isIndexExist(String indexName, ActionListener<AnomalyLocalizationOutput> listener) {
+        String[] concreteIndices = indexNameExpressionResolver.concreteIndexNames(clusterService.state(),
+                IndicesOptions.lenientExpandOpen(), indexName);
+        if (concreteIndices == null || concreteIndices.length == 0) {
+            return false;
+        }
+        for(String index : concreteIndices) {
+            if (!clusterService.state().metadata().hasIndex(index)) {
```

Comment:
If "logs*" pattern has 3 concrete index "logs1", "logs2" and "logs3", if "log1" doesn't exist but "log2" and "log3" exist, we should return true as we still have 2 valid index exist. But this code seems will return false unless all three indexes exist.

Reply:
ok, I was confused about this use case. Updated the logic to return true if any valid index exist.